extensions/parquet/table/plugin/channelprovider/s3/src/main/java/io/deephaven/parquet/table/plugin/channelprovider/s3/S3SeekableChannelProvider.java

```diff
@@ -60,7 +58,7 @@ public SeekableByteChannel getWriteChannel(@NotNull final Path path, final boole
         throw new UnsupportedOperationException("Don't support writing to S3 yet");
     }
 
-    public void close() throws IOException {
+    public void close() {
```

Comment:
Nothing calls this currently; do we want them to? Should this class implement Closeable?

Reply:
Need help for this one, no class really owns a channel provider object, its being passed around between parquet file reader objects. So not sure when to close it.

Reply:
This is a good catch, Shivam. The model this code is creating is one of "long lived" connections.
1. We probably need some sort of support to re-open a connection if it's "lost". Automatically.
2. We might want to cache connections for re-use. Maybe we want one per region, only?
3. Maybe we want a timeout, after which we close unused connections? (This would play nicely with (1)).

Reply:
Arguably, 1 and 3 are enough for robustness, but I worry that 2 is necessary if AWS imposes limited on the number of outstanding connections.